counterexamples/direct_sum_is_internal.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser, Kevin Buzzard
+-/
+
+import algebra.direct_sum.module
+import tactic.fin_cases
+
+/-!
+# Not all complementary decompositions of a module make up a direct sum
+
+This shows that while `ℤ≤0` and `ℤ≥0` are complementary `ℕ`-submodules of `ℤ`, which in turn
+implies as a collection they are `complete_lattice.independent` and that they span all of `ℤ`, they
+do not permit a decomposition into a direct sum.
+
+This file demonstrates why `direct_sum.submodule_is_internal_of_independent_of_supr_eq_top` must take
+`ring R` and not `semiring R`.
+-/
+
+lemma units_int.one_ne_neg_one : (1 : units ℤ) ≠ -1 :=
```

Comment:
Can `dec_trivial` not see this? Also, this should move to another file, I guess.

Reply:
We don't have this as a lemma for integers either, so I'm not convinced it's worthy of being a lemma in the library. `dec_trivial` almost certainly does work; do you want me to change it?

Reply:
Ok, let's leave it like this then

Reply:
I changed it to `dec_trivial` since there was a cache available to test with.